.github/workflows/step_test.yaml

```diff
@@ -1,52 +1,50 @@
 name: test
-run-name: Run tests
-
-concurrency:
-  group: ${{ github.ref }}
-  cancel-in-progress: true
 
 on:
   workflow_call:
+    inputs:
+      mask-experimental:
+        type: boolean
+        default: true
+        description: Always report experimental test as successful
 
 permissions:
   contents: read
 
 jobs:
-  pre-commit:
-    name: Check pre-commit
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v4
-        with:
-          python-version: 3.x
-      - uses: pre-commit/action@v3.0.0
 
   tests:
-    name: Check ${{ matrix.toolchain }} and Python ${{ matrix.python-version }}
+    name: >
+      🖥️ ${{ matrix.os || 'Fedora' }} +
+      ${{ !matrix.os && format('🛠️ {0} +', matrix.toolchain) || '' }}
+      🐍 ${{ matrix.python-version }}
+      ${{ matrix.experimental && '[🧪 Experimental]' || '' }}
     runs-on: ${{ matrix.os || 'ubuntu-latest' }}
     container: ${{ !matrix.os && 'ghcr.io/lecrisut/dev-env:main' || '' }}
-    needs: [ pre-commit ]
     continue-on-error: ${{ matrix.experimental || false }}
     strategy:
       fail-fast: false
       matrix:
-        toolchain: [ gcc, llvm, intel, windows, macos ]
-        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12"]
+        toolchain: [ gcc, llvm, intel ]
+        python-version: [ "3.8", "3.x" ]
```

Comment:
Note: "3.x" specifies the latest stable version.
https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-the-python-version-input